db/redis.js

```diff
@@ -116,8 +116,17 @@ async function addVitals(locationid, signalStrength, cloudDisconnects) {
   client.xadd(`vitals:${locationid}`, 'MAXLEN', '~', '10000', '*', 'strength', signalStrength, 'cloudDisc', cloudDisconnects)
 }
 
-async function addEdgeDeviceHeartbeat(locationid, doorStatus, doorTime, insTime) {
-  client.xadd(`heartbeat:${locationid}`, 'MAXLEN', '~', '10000', '*', 'doorStatus', doorStatus, 'doorTime', doorTime, 'insTime', insTime)
+// ignore comments included to allow arguments to be split across lines in pairs
+// prettier-ignore
+/* eslint-disable function-call-argument-newline */
+async function addEdgeDeviceHeartbeat(locationid, missedDoorMessages, doorLowBattery, doorHeartbeatReceived, stateTransitions) {
+  client.xadd(
+    `heartbeat:${locationid}`, 'MAXLEN', '~', '10000', '*',
+    'missedDoorMessages', missedDoorMessages,
+    'doorLowBattery', doorLowBattery,
+    'doorHeartbeatReceived', doorHeartbeatReceived,
+    'stateTransitions', stateTransitions
```

Comment:
I left the state transitions array as one piece of a big redis entry rather than multiple redis entries

Reply:
OK.

Reply:
What does this look like in Redis?
Is the `client.xadd` function smart enough to serialize the `stateTransitions` object into something pretty? 
Or should you use `JSON.stringify(stateTransitions)`?

Reply:
After some testing, it turns out redis won't accept arrays as an value. I will need to stringify